proto/regen/ecocredit/v1beta1/state.proto

```diff
@@ -0,0 +1,213 @@
+syntax = "proto3";
+
+package regen.ecocredit.v1beta1;
+
+import "cosmos/orm/v1alpha1/orm.proto";
+import "cosmos/base/v1beta1/coin.proto";
+import "gogoproto/gogo.proto";
+import "google/protobuf/timestamp.proto";
+
+// ClassInfo represents the high-level on-chain information for a credit class.
+message ClassInfo {
+
+    option (cosmos.orm.v1alpha1.table) = {
+        id: 1,
+        primary_key: {fields: "class_id"}
+    };
+
+    // class_id is the unique ID of credit class.
+    string class_id = 1;
+
+    // admin is the admin of the credit class.
+    string admin = 2;
+
+    // issuers are the approved issuers of the credit class.
+    repeated string issuers = 3;
+
+    // metadata is any arbitrary metadata to attached to the credit class.
+    bytes metadata = 4;
+
+    // credit_type describes the type of credit (e.g. carbon, biodiversity), as well as unit and precision.
+    CreditType credit_type = 5;
+
+    // The number of batches issued in this credit class.
+    uint64 num_batches = 6;
+}
+
+// ProjectInfo represents the high-level on-chain information for a project.
+message ProjectInfo {
+
+    option (cosmos.orm.v1alpha1.table) = {
+        id: 2,
+        primary_key: {fields: "project_id"}
+    };
+
+    // project_id is the unique ID of the project.
+    string project_id = 1;
+
+    // class_id is the unique ID of credit class for this project.
+    string class_id = 2;
+
+    // issuer is the issuer of the credit batches for this project.
+    string issuer = 3;
+
+    // project_location is the location of the project.
+    // Full documentation can be found in MsgCreateProject.project_location.
+    string project_location = 4;
+
+    // metadata is any arbitrary metadata attached to the project.
+    bytes metadata = 5;
+}
+
+// BatchInfo represents the high-level on-chain information for a credit batch.
+message BatchInfo {
+
+    option (cosmos.orm.v1alpha1.table) = {
+        id: 3,
+        primary_key: {fields: "batch_denom"}
```

Comment:
Let's actually add an uint64 auto increment primary key and make this an unique index. I think we'll want a short uint64 to reference batches in the order book and denoms are getting kind of long

Reply:
are there docs anywhere for the syntax on this?

Reply:
in orm.proto
```suggestion
        primary_key: {fields: "id", auto_increment: true}
        index: {fields: "batch_denom", unique: true}
```